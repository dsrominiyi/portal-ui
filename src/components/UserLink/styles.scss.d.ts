export const userLink: string;
export const userContainer: string;
export const userName: string;
export const topbarDivider: string;
