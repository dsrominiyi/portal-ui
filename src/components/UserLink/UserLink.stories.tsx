import React from 'react';
import { storiesOf } from '@storybook/react';

import UserLink from '.';

const stories = storiesOf('UserLink', module);

const render = (name: UserName, imageUrl?: string) => () => (
  <UserLink name={name} imageUrl={imageUrl} />
);

const name = { forename: 'John', surname: 'Smith' };

export const storyRenders = {
  noPhoto: render(name),
  withPhoto: render(name, '/images/person-who-does-not-exist.jpg')
};

stories.add('With no profile photo', storyRenders.noPhoto);
stories.add('With a profile photo', storyRenders.withPhoto);
