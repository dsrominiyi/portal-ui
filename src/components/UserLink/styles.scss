@import '../../style/vars';

.user-link {
  display: inline-flex;
  align-items: center;
  justify-content: center;
}

.user-container {
  display: inline-flex;
  align-items: center;
  cursor: pointer;

  .user-name {
    color: $brand-mid-grey;
    margin-right: 1rem;
    font-size: 80%;
    font-weight: 400;
  }
}

.topbar-divider {
  width: 0;
  height: 2.375rem;
  border-right: 1px solid rgb(227, 230, 240);
  margin: 0 1rem 0 0;
  display: inline-block;
}
