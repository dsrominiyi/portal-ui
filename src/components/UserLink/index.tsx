import React, { FunctionComponent } from 'react';

import UserImage from '../UserImage';

import styles from './styles.scss';

interface Props {
  name: UserName;
  imageUrl?: string;
}

export const UserLink: FunctionComponent<Props> = ({ name, imageUrl }: Props) => (
  <div className={styles.userLink}>
    <div className={styles.topbarDivider} />
    <div className={styles.userContainer}>
      <span className={styles.userName}>{`${name.forename} ${name.surname}`}</span>
      <UserImage name={name} imageUrl={imageUrl} />
    </div>
  </div>
);

UserLink.displayName = 'UserLink';
export default UserLink;
