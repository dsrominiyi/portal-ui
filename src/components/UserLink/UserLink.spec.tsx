import { storyRenders } from './UserLink.stories';

import testStorySnapshots from '../../helpers/test/testStorySnapshots';

describe('IconLink', () => {
  test('component story renders match snapshots', () => {
    testStorySnapshots(storyRenders);
  });
});
