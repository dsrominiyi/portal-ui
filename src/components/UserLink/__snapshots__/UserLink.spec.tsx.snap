// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconLink component story renders match snapshots: noPhoto 1`] = `
<div
  className="userLink"
>
  <div
    className="topbarDivider"
  />
  <div
    className="userContainer"
  >
    <span
      className="userName"
    >
      John Smith
    </span>
    <UserImage
      name={
        Object {
          "forename": "John",
          "surname": "Smith",
        }
      }
    />
  </div>
</div>
`;

exports[`IconLink component story renders match snapshots: withPhoto 1`] = `
<div
  className="userLink"
>
  <div
    className="topbarDivider"
  />
  <div
    className="userContainer"
  >
    <span
      className="userName"
    >
      John Smith
    </span>
    <UserImage
      imageUrl="/images/person-who-does-not-exist.jpg"
      name={
        Object {
          "forename": "John",
          "surname": "Smith",
        }
      }
    />
  </div>
</div>
`;
